arrows/ffmpeg/ffmpeg_video_output.cxx

```diff
@@ -484,23 +485,56 @@ ::add_image( kv::image_container_sptr const& image,
   // Fill in a few mandatory fields
   frame->width = image->width();
   frame->height = image->height();
-  frame->format = AV_PIX_FMT_RGB24;
+  switch( image->depth() )
+  {
+    case 1: frame->format = AV_PIX_FMT_GRAY8; break;
+    case 3: frame->format = AV_PIX_FMT_RGB24; break;
+    default:
+      VITAL_THROW( kv::invalid_value,
+                   "Image has unsupported depth: " +
+                   std::to_string( image->depth() ) );
+  }
+
+  if( image->get_image().pixel_traits() !=
+      kv::image_pixel_traits_of< uint8_t >() )
+  {
+    // TODO: Is there an existing conversion function somewhere?
```

Comment:
@mleotta Might know the answer to this. Otherwise, LGTM

Reply:
I'll investigate and add this in a future branch